### PR TITLE
Add selector tests

### PR DESCRIPTION
**Pull request by Zirro**
These tests cover a few issues found during the development of [nwsapi](https://github.com/dperini/nwsapi) (used in [jsdom](https://github.com/jsdom/jsdom)) that were not caught by the existing tests.

### CODE REVIEW

File: dom/nodes/selectors.js [hunk 1]
```diff
@@ -536,10 +538,11 @@ var scopedSelectors = [
...
+  {name: ":nth-child selector, matching every third child element",                            selector: "li:nth-child(3n)",            ctx: "#pseudo-nth", expect: ["pseudo-nth-li3", "pseudo-nth-li6", "pseudo-nth-li9", "pseudo-nth-li12"],                    level: 3, testType: TEST_FIND | TEST_MATCH},
+  {name: ":nth-child selector, matching every second child element, starting from the fourth", selector: "li:nth-child(2n+4)",          ctx: "#pseudo-nth", expect: ["pseudo-nth-li4", "pseudo-nth-li6", "pseudo-nth-li8", "pseudo-nth-li10", "pseudo-nth-li12"], level: 3, testType: TEST_FIND | TEST_MATCH},
+  {name: ":nth-child selector, matching every fourth child element, starting from the third",  selector: ":nth-child(4n-1)",            ctx: "#pseudo-nth-p1", expect: ["pseudo-nth-em2", "pseudo-nth-span3"],                                                    level: 3, testType: TEST_FIND | TEST_MATCH},
```
Comment (annevk) [inline dom/nodes/selectors.js]: The above 4 lines are only whitespace changes?

**Review by annevk — COMMENTED** [review]
Thanks! Did you ran these in browsers too? If there's no failures there I'm happy with landing this.

File: dom/nodes/selectors.js (hunk 1 above)
Comment (Zirro) [reply dom/nodes/selectors.js]: Yes, for the properties to remain aligned with each other.

**Comment by Zirro** [comment]
> Did you ran these in browsers too? If there's no failures there I'm happy with landing this.

Yeah, these tests are passing in Chrome, Firefox and Safari.

**Review by annevk — APPROVED** [review]

**Pull request merged**